app/components/SignInSignOut.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { Component, PropTypes } from 'react';
+import Spinner from 'react-mdl/lib/Spinner';
+import Constants from '../constants';
+import AuthActions from '../actions/auth';
+
+export default class SignInSignOut extends Component {
+  render() {
+    const { dispatch, auth } = this.props;
+    let content;
+
+    switch (auth.state) {
```

Comment:
We probably don't need "sign in" link here. When user is signed out, they'll be redirected to the landing page and thus never see this drawer.